File: Array/2210. Count Hills and Valleys in an Array.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums) 
    {
        int ans = 0, n = nums.size();
        
        vector<int> arr = {nums[0]};
        for(int i = 1; i < n; i++)
        {
            if(nums[i] != nums[i - 1])
                arr.push_back(nums[i]);
        }

        nums = arr;
        n = nums.size();
        for(int i = 1; i < n - 1; i++)
        {
            ans += (nums[i] > nums[i + 1] and nums[i] > nums[i - 1]);
            ans += (nums[i] < nums[i + 1] and nums[i] < nums[i - 1]);
        }
        return ans;
    }
};

File: Array/2210. Count Hills and Valleys in an Array.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        temp = [nums[0]]
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                temp.append(nums[i])
        nums = temp

        ans = 0
        for i in range(1, len(nums) - 1):
            ans += (nums[i - 1] < nums[i] and nums[i + 1] < nums[i])
            ans += (nums[i - 1] > nums[i] and nums[i + 1] > nums[i])
        return ans